tile-group.lisp

```diff
@@ -1315,9 +1315,50 @@ direction. The following are valid directions:
         (message "There's only one frame.")
         (balance-frames-internal group tree))))
 
+(defun calculate-window-centerpoint (win)
+  "window x and y are calculated from the top of the screen. thus the x/y are the upper left hand 
+corner. to find the center x/y we do: xc = x + (w ÷ 2) and yc = y + (h ÷ 2)"
+  (let ((x (window-x win))
+        (y (window-y win))
+        (w (window-width win))
+        (h (window-height win)))
+    (let ((center-x (+ x (/ w 2)))
+          (center-y (+ y (/ h 2))))
+      (list center-x center-y))))
+
+(defun calculate-frame-centerpoint (frame)
+  "Calculates the centerpoint of a frame, not including borders. "
+  (let ((x (frame-x frame))
+        (y (frame-y frame))
+        (w (frame-width frame))
+        (h (frame-height frame)))
+    (let ((center-x (+ x (/ w 2)))
+          (center-y (+ y (/ h 2))))
+      (list center-x center-y))))
+
+(defun calculate-appropriate-frame-for-window (&optional (win (current-window))
```

Comment:
Instead of the generic word appropriate it is better to say what is the specific criteria we are using for selecting the frame. In this case it is the `closest-frame`.

There is no reason for the arguments to be `&optional`.